### ISSUE TITLE
Feature request: Rule to prevent unused/incorrect example data

### ISSUE THREAD

**Issue by DwordPtr** [vsiakka/gherkin-lint#89]
I lost some time yesterday due to a tab header in an example set being different after a 'Step_arg` name change. It occurs to me this is the kind of problem a computer should solve instead of a human and it shouldn't (might) not be too hard to write a rule to check if there are any example headings not needed/used in the scenario above.

I need help because I haven't dealt with parsers and don't know what the js objects look like that burp outputs. I've tried looking at the docs but I haven't been able to find what the output objects look like.

I totally don't mind putting some time into this because it would be awesome, but I'm a bit of a noob so I could use some help :P

**Comment by vsiakka** [comment]
Hey @DwordPtr, is this type of error you would like the linter to catch?
Feature: missmatched example variables

Scenario Outline: test
  Given <a> doesn't exist

Examples:
|v|
|1|``

**Comment by DwordPtr** [comment]
Yes

**Comment by vsiakka** [comment]
Sorry Bryan, I didn't read the last part of your message carefully enough and I didn't realize you wanted to implement this yourself, till I put up the PR and came back here to ping you about it T_T

The way I have written most of these rules is by console logging a [parsed file](https://github.com/vsiakka/gherkin-lint/blob/master/src/linter.js#L19) and seeing its structure. 
The gherkin parser (we use the official gherkin parser package) has a set of [examples](https://github.com/cucumber/cucumber/tree/master/gherkin/testdata/good), but I find it easier to writte my own feature files, console.log the generated ast, edit the file etc.

For this rule, I also looked around the [gherkin source](https://github.com/cucumber/cucumber/tree/e88ca59d72fac9a4a4899b0b5f67732c908b39ea/gherkin/javascript/lib/gherkin), to validate some assumptions, mostly about whether some things can be undefined, and that wasn't super tricky (I started by searching the repo for ScenarioOutlines and worked my way backwards).

Regarding the PR i put up, feel free to take a look. I want to review it again before merging with a clearer mind.
